src/dms/core/EdFi.DataManagementService.Core/ApiSchema/ApiSchemaDocuments.cs

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Text.Json.Nodes;
+using EdFi.DataManagementService.Core.ApiSchema.Model;
+using EdFi.DataManagementService.Core.External.Model;
+using Microsoft.Extensions.Logging;
+
+namespace EdFi.DataManagementService.Core.ApiSchema;
+
+/// <summary>
+/// Provides information from loaded ApiSchema.json documents
+/// </summary>
+internal class ApiSchemaDocuments(ApiSchemaNodes _apiSchemaNodes, ILogger _logger)
+{
+    /// <summary>
+    /// Gets the core ProjectSchema node in the document.
+    /// </summary>
+    public ProjectSchema GetCoreProjectSchema()
+    {
+        JsonNode projectSchemaNode =
+            _apiSchemaNodes.CoreApiSchemaRootNode["projectSchema"]
+            ?? throw new InvalidOperationException("Expected projectSchema node to exist.");
+        return new ProjectSchema(projectSchemaNode, _logger);
+    }
+
+    /// <summary>
+    /// Gets the extension ProjectSchemas.
+    /// </summary>
+    public ProjectSchema[] GetExtensionProjectSchemas()
+    {
+        return _apiSchemaNodes
+            .ExtensionApiSchemaRootNodes.Select(node =>
+                node["projectSchema"]
+                ?? throw new InvalidOperationException("Expected projectSchema node to exist.")
+            )
+            .Select(node => new ProjectSchema(node, _logger))
+            .ToArray();
+    }
+
+    /// <summary>
+    /// Finds the ProjectSchema that represents the given ProjectNamespace e.g. "ed-fi" for the Data Standard.
+    /// Returns null if not found.
+    /// </summary>
+    public ProjectSchema? FindProjectSchemaForProjectNamespace(ProjectNamespace projectNamespace)
```

Comment:
Code needed lookup by both ProjectNamespace and ProjectName - branded types pointed this out.